Makefile

```
@@ -167,7 +167,7 @@ test: gotestsum ## Run tests.
 .PHONY: test-integration
 test-integration: gomod-download envtest ginkgo mpi-operator-crd ray-operator-crd jobset-operator-crd kf-training-operator-crd  cluster-autoscaler-crd ## Run tests.
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" \
-	$(GINKGO) $(GINKGO_ARGS) --junit-report=junit.xml --output-dir=$(ARTIFACTS) -v $(INTEGRATION_TARGET)
+	$(GINKGO) $(GINKGO_ARGS) -procs=4 --junit-report=junit.xml --output-dir=$(ARTIFACTS) -v $(INTEGRATION_TARGET)
```

```suggestion
	$(GINKGO) $(GINKGO_ARGS) -procs=$(INTEGRATION_NPROCS) --junit-report=junit.xml --output-dir=$(ARTIFACTS) -v $(INTEGRATION_TARGET)
```

Should we make a parameter for the number of processes?

And then, I think that we can have the `INTEGRATION_NPROCS ?= $(shell nproc)` var.

---

According to https://onsi.github.io/ginkgo/#spec-parallelization `-p` will do that.

While running in prow the number of go procs is limited:

https://github.com/kubernetes/test-infra/blob/21fe7ceb7858ecaf0fb9ab38c5222b0b0e2de8bd/config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml#L50-L60

But if there is chance that 4 resource cpu's can translate in a timeshare of more then 4 we can drop GOMAXPROCS env setting.

Another thing that could be useful is to still be able to run it non-parallel if needed.

---

Yes, I'll create a parameter. I prefer to hardcode the value. During testing, I encountered file descriptors exhausting when I set a high-number of processors (see end of [comment](https://github.com/kubernetes-sigs/kueue/issues/1737#issuecomment-1956509955)). I just tested now and I encountered this on my machine when using the value returned by nproc (24).

---

I'm ok with making it default to 4 (not the number of processes), due to the issue with file descriptors which would also probably affect my machine. The parameter still allows developers to tune the value if they have more cores.

---

Thank you for the clarifications. I'm ok with making it 4 as a default.